adapter/dubbogo/consumer_filter.go

```diff
@@ -0,0 +1,50 @@
+package dubbogo
+
+import (
+	"github.com/apache/dubbo-go/protocol"
+)
+import (
+	sentinel "github.com/alibaba/sentinel-golang/api"
+	"github.com/alibaba/sentinel-golang/core/base"
+)
+
+type consumerFilter struct{}
+
+func (d *consumerFilter) Invoke(invoker protocol.Invoker, invocation protocol.Invocation) protocol.Result {
+	methodResourceName := getResourceName(invoker, invocation, getConsumerPrefix())
+	interfaceResourceName := ""
+	if getInterfaceGroupAndVersionEnabled() {
+		interfaceResourceName = getColonSeparatedKey(invoker.GetUrl())
+	} else {
+		interfaceResourceName = invoker.GetUrl().Service()
+	}
+	if !isAsync(invocation) {
+		_, b := sentinel.Entry(interfaceResourceName,
+			sentinel.WithResourceType(base.ResTypeRPC),
+			sentinel.WithTrafficType(base.Outbound))
+		if b != nil { // blocked
+			result := &protocol.RPCResult{}
+			result.SetResult(nil)
+			result.SetError(b)
+			return result
+		}
+		_, b = sentinel.Entry(methodResourceName,
+			sentinel.WithResourceType(base.ResTypeRPC),
+			sentinel.WithTrafficType(base.Outbound),
+			sentinel.WithArgs(invocation.Attachments()))
+		if b != nil { // blocked
+			result := &protocol.RPCResult{}
```

Comment:
Can we also provide a fallback registry where developers could register their own fallback logic when blocked? We could provide the default behavior.

Refer: https://github.com/alibaba/Sentinel/tree/1.7.1/sentinel-adapter/sentinel-apache-dubbo-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/dubbo/fallback

Reply:
please check it